src/business-logic/services/database-change-ingest-service.ts

```diff
@@ -0,0 +1,283 @@
+    this.eventQueue.push(event)
+    this.executeNextEvent()
+  }
+
+  private executeNextEvent(): void {
+    if (this.isExecutingEvent) {
+      return
+    }
+    if (Date.now() - this.lastBulkExecutionStartTimestamp >= BULK_EXECUTION_TIMESPAN_IN_MS) {
+      this.lastBulkExecutionStartTimestamp = Date.now()
+    }
+    const eventCallback: (() => Promise<void>) | undefined = this.eventQueue.shift()
+    if (!eventCallback) {
+      return
+    }
+
+    this.isExecutingEvent = true
+    eventCallback()
+      .catch(error => console.error('Error when executing Ingest event:', error))
+      .finally(() => {
+        this.isExecutingEvent = false
+        void this.buildRundowns()
+        this.executeNextEvent()
+      })
+  }
+
+  private async buildRundowns(): Promise<void> {
+    try {
+      if (Date.now() - this.lastBulkExecutionStartTimestamp < BULK_EXECUTION_TIMESPAN_IN_MS && this.eventQueue.length > 0) {
+        return
+      }
+
+      for (const rundownId of this.rundownIdsToBuild.values()) {
+        const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
+        if (!rundown.isActive()) {
+          continue
+        }
+        await this.buildAndPersistTimeline(rundown)
+        this.emitSetNextEvent(rundown)
+      }
+      this.rundownIdsToBuild.clear()
+    } catch (error) {
+      console.log('Error when trying to build Rundowns for bulk', error)
+    }
+  }
+
+  private async buildAndPersistTimeline(rundown: Rundown): Promise<void> {
+    const timeline: Timeline = await this.timelineBuilder.buildTimeline(rundown)
+    await this.timelineRepository.saveTimeline(timeline)
+  }
+
+  private emitSetNextEvent(rundown: Rundown): void {
+    const setNextEvent: PartSetAsNextEvent = this.eventBuilder.buildSetNextEvent(rundown)
+    this.eventEmitter.emitRundownEvent(setNextEvent)
+  }
+
+  private async createRundown(rundown: Rundown): Promise<void> {
+    const rundownCreatedEvent: RundownCreatedEvent = this.eventBuilder.buildRundownCreatedEvent(rundown)
+    this.eventEmitter.emitRundownEvent(rundownCreatedEvent)
+    await this.rundownRepository.saveRundown(rundown)
+  }
+
+  private async updateRundown(rundown: Rundown): Promise<void> {
+    const rundownUpdatedEvent: RundownUpdatedEvent = this.eventBuilder.buildRundownUpdatedEvent(rundown)
+    this.eventEmitter.emitRundownEvent(rundownUpdatedEvent)
+    await this.rundownRepository.saveRundown(rundown)
+  }
+
+  private async deleteRundown(rundownId: string): Promise<void> {
+    const rundownDeletedEvent: RundownDeletedEvent = this.eventBuilder.buildRundownDeletedEvent(rundownId)
+    this.eventEmitter.emitRundownEvent(rundownDeletedEvent)
+    await this.rundownRepository.deleteRundown(rundownId)
+  }
+
+  private async createSegment(segment: Segment): Promise<void> {
+    const rundown: Rundown = await this.rundownRepository.getRundown(segment.rundownId)
+    try {
+      rundown.addSegment(segment)
+    } catch (error) {
```

Comment:
There is a inconsistency in the naming of caught errors. Here error is used. Further down it is exception.

Reply:
Updated it so all try-catch for the "events" calls them `exception`. I left it as `error` for `executNextEvent()` and `buildRundowns()` because they have nothing "exception" specific going on.
Let me know if I should rename those as well.